mantidproject/mantid
# Engg GUI: support event mode

Fixes #13808, introduces basic pre-processing (re-binning) capability for event mode runs.

**To test**:

You'll need an example ENGINX event mode run file (available from the archive, cycle 12_2): ENGINX00197019. This is a multi-period run (with two periods).

Check that the documentation gives a good idea of what the new tab currently can do.

Open Interfaces->Diffraction->Engineering diffraction
Change to the tab "Pre-processing"
For the run #, browse to the file ENGINX00197019.nxs (use the nexus one, as the raw is already histogrammed!)

In the "regular time binning" group:
- You can leave the default time bin (0.1s), that's will be not to slow
- Click rebin. This should produce the input and output workspaces, visible in the ADS
- for this file the workspaces are groups of two (it's a multiperiod run with two periods)

In the "multi-period data (rebinning by pulse time)" box
- You can leave the default settings
- Click rebin
- See that the x axis of the output workspace(s) are time - (not TOF as in the normal rebinning)

In all cases the output workspaces should look fine. They should have
the 2400 spectra (both banks). etc.
- The user inputs/options should be remembered between sessions.
- If you try to pre-process a file of a wrong type (other ENGINX files), it should still be able to produce some output workspace or fail gracefully.
- a few test cases have been added. A couple of them could be disabled if they slow down this GUI's unit test too much:
  - test_preproc_event_time_ok()
  - test_preproc_event_multiperiod_file_wrong_type()
    (they load a small file and start the rebin; the first one runs the rebin)

Note: The EnggDiffWorker class badly needs a bit of refactoring to split it into the individual workers. That will follow in another issue.
